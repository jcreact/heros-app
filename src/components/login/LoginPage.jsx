import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import {
    Card,
    CardContent,
    CardMedia,
    TextField,
    FormControl,
    Box,
    Button,
    Grid,
    CssBaseline,
} from '@material-ui/core';
import { ThemeProvider } from '@material-ui/core/styles';
import { darkTheme } from '../../theme/themes';

const useStyles = makeStyles((theme) => ({
    main: {
        height: '90vh',
    },
    card: {
        width: 300,
    },
    media: {
        paddingTop: '30%',
        marginInline: theme.spacing(1),
        marginTop: theme.spacing(1),
    },
    field: {
        marginBottom: theme.spacing(2),
    },
}));

export const LoginPage = ({ history }) => {
    const classes = useStyles();

    const handleSubmit = (evt) => {
        evt.preventDefault();
        history.replace('/');
    };

    return (
        <ThemeProvider theme={darkTheme}>
            <CssBaseline />
            <Grid
                container
                direction="row"
                justify="center"
                alignItems="center"
                className={classes.main}
            >
                <form onSubmit={handleSubmit}>
                    <Card className={classes.card} variant="outlined">
                        <CardMedia
                            className={classes.media}
                            image="./assets/login.png"
                        />
                        <CardContent>
                            <FormControl fullWidth className={classes.field}>
                                <TextField
                                    label="Usuario"
                                    autoFocus
                                    autoComplete="off"
                                    value="test"
                                />
                            </FormControl>
                            {/* <FormControl fullWidth className={classes.field}>
                                <TextField
                                    type="password"
                                    label="ContraseÃ±a"
                                    autoComplete="off"
                                    value="123456"
                                />
                            </FormControl> */}
                        </CardContent>
                        <Box display="flex" justifyContent="center" mb={3}>
                            <Button
                                variant="contained"
                                color="primary"
                                type="submit"
                            >
                                Ingresar
                            </Button>
                        </Box>
                    </Card>
                </form>
            </Grid>
        </ThemeProvider>
    );
};
