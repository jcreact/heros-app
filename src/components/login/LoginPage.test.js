import { mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

// Componente a probar.
import { LoginPage } from './LoginPage';

describe(`Pruebas en el componente 'LoginPage.jsx'.`, () => {

    const contextValue = {
        dispatch: jest.fn()
    };

    const history = {
        replace: jest.fn()
    };



    const component = mount(
        <AuthContext.Provider value={contextValue}>
            <LoginPage history={history} />
        </AuthContext.Provider>
    );

    test(`Debe renderizar el componente correctamente.`, () => {
        expect(component).toMatchSnapshot();
    });

    test(`Debe hacer el dispatch (login) y hacer el replace al '/'.`, () => {

        const action = {
            type: types.login,
            payload: {
                logged: true,
                name: 'JosÃ© Palma'
            }
        };

        component.find('form').prop('onSubmit')({ preventDefault() { } });
        expect(contextValue.dispatch).toHaveBeenCalledWith(action);
        expect(history.replace).toHaveBeenCalledWith('/');
    });

    test(`Debe hacer el replace al '/dc'.`, () => {

        localStorage.setItem('lastPath', '/dc');
        const component = mount(
            <AuthContext.Provider value={contextValue}>
                <LoginPage history={history} />
            </AuthContext.Provider>
        );

        component.find('form').prop('onSubmit')({ preventDefault() { } });
        expect(history.replace).toHaveBeenCalledWith('/dc');

    });
});
