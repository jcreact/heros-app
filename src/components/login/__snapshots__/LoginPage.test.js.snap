// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente 'LoginPage.jsx'. Debe renderizar el componente correctamente. 1`] = `
Array [
  "",
  <div
    className="MuiGrid-root makeStyles-main-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="MuiPaper-root MuiCard-root makeStyles-card-2 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          className="MuiCardMedia-root makeStyles-media-3"
          style={
            Object {
              "backgroundImage": "url(\\".\\")",
            }
          }
        />
        <div
          className="MuiCardContent-root"
        >
          <div
            className="MuiFormControl-root makeStyles-field-4 MuiFormControl-fullWidth"
          >
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink={true}
              >
                Usuario
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoComplete="off"
                  autoFocus={true}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value="jcpalma"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiBox-root MuiBox-root-5"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Ingresar
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>,
]
`;
